tests/unit/test_utils.py

```diff
@@ -157,3 +157,33 @@ def test_human_time_negative_float(data: HumanTimeTestData) -> None:
     """
     result = utils.human_time(-float(data.value))
     assert result == f"-{data.expected}"
+
+
+class RoundHalfUpTestData(NamedTuple):
+    """Data for round half up tests."""
+
+    id: str
+    value: Union[int, float]
+    expected: int
+
+
+round_half_up_test_data = [
+    RoundHalfUpTestData(id="integer", value=1, expected=1),
+    RoundHalfUpTestData(id="negative integer", value=-1, expected=-1),
+    RoundHalfUpTestData(id="down float", value=1.49999999, expected=1),
+    RoundHalfUpTestData(id="up float", value=1.50000001, expected=2),
+    RoundHalfUpTestData(id="negative down float", value=-1.49999999, expected=-1),
+    RoundHalfUpTestData(id="negative up float", value=-1.50000001, expected=-2),
+]
+
+
+@pytest.mark.parametrize("data", round_half_up_test_data, ids=lambda data: data.id)
+def test_round_half_up(data: RoundHalfUpTestData) -> None:
+    """Test for the utils.round_half_up function.
+
+    Ensure the number passed is consistently rounded to the nearset
+    integer with ties going away from zero.
+    """
+    for _iteration in range(100):
```

Comment:
I don't understand the point of this loop. Why is it here? Are you testing that calling the function more than once gives the same result? I don't really get it...

Reply:
Ha, the intention was to demonstrate the consistency of the rounding technique vs. round(), but this doesn't even do that :) I'll remove the loop and make a note